controllers/space/space_controller_test.go

```diff
@@ -42,228 +42,259 @@ func TestCreateSpace(t *testing.T) {
-				nsTmplSet.Status.Conditions = []toolchainv1alpha1.Condition{
-					nstemplatetsettest.Provisioned(),
-				}
-				nsTmplSet.Status.ProvisionedNamespaces = []toolchainv1alpha1.SpaceNamespace{
-					{
-						Name: "john-dev",
-						Type: "default",
-					},
-					{
-						Name: "john-stage",
-					},
+		tests := []testRun{
+			{
+				name: "without feature toggles",
+			},
+			{
+				name:           "with feature toggles",
+				featureToggles: "feature-1,feature-2",
+			},
+		}
+		for _, testRun := range tests {
+			t.Run(testRun.name, func(t *testing.T) {
+				spaceOptions := []spacetest.Option{spacetest.WithSpecTargetCluster("member-1")}
+				if testRun.featureToggles != "" {
+					spaceOptions = append(spaceOptions, spacetest.WithAnnotation(toolchainv1alpha1.FeatureToggleNameAnnotationKey, testRun.featureToggles))
 				}
-				err := member1.Client.Update(context.TODO(), nsTmplSet)
-				require.NoError(t, err)
+
+				// given
+				s := spacetest.NewSpace(test.HostOperatorNs, "oddity", spaceOptions...)
+				hostClient := test.NewFakeClient(t, s, base1nsTier)
+				member1 := NewMemberClusterWithTenantRole(t, "member-1", corev1.ConditionTrue)
+				member2 := NewMemberClusterWithTenantRole(t, "member-2", corev1.ConditionTrue)
+				InitializeCounters(t,
+					NewToolchainStatus())
 				ctrl := newReconciler(hostClient, member1, member2)
 
 				// when
-				_, err = ctrl.Reconcile(context.TODO(), requestFor(s))
+				res, err := ctrl.Reconcile(context.TODO(), requestFor(s))
 
 				// then
 				require.NoError(t, err)
-				assert.Equal(t, reconcile.Result{Requeue: false}, res) // no requeue and status with new provisioned namespaces was updated
+				assert.True(t, res.Requeue) // requeue requested explicitly when NSTemplateSet is created, even though watching the resource is enough to trigger a new reconcile loop
 				spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
 					Exists().
 					HasStatusTargetCluster("member-1").
-					HasStatusProvisionedNamespaces(nsTmplSet.Status.ProvisionedNamespaces).
-					HasConditions(spacetest.Ready()). // space is still provisioning
+					HasConditions(spacetest.Provisioning()).
 					HasStateLabel("cluster-assigned").
 					HasFinalizer()
+				var nsTmplSet *toolchainv1alpha1.NSTemplateSet
+				nsTmplSetAssertion := nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, "oddity", member1.Client).
+					Exists().
+					HasTierName(base1nsTier.Name).
+					HasClusterResourcesTemplateRef("base1ns-clusterresources-123456new").
+					HasNamespaceTemplateRefs("base1ns-code-123456new", "base1ns-dev-123456new", "base1ns-stage-123456new")
+				if testRun.featureToggles != "" {
+					nsTmplSetAssertion = nsTmplSetAssertion.
+						HasAnnotationWithValue(toolchainv1alpha1.FeatureToggleNameAnnotationKey, testRun.featureToggles)
+				} else {
+					nsTmplSetAssertion = nsTmplSetAssertion.
+						DoesNotHaveAnnotation(toolchainv1alpha1.FeatureToggleNameAnnotationKey)
+				}
```

Comment:
This is the second change. Check the NsTemplateSet feature annotation if it's present in the Space.
The rest of this test is not changed and just formatted to be moved to the table test wrapper.